source/server/admin/profiling_handler.cc

```diff
@@ -76,5 +76,18 @@ Http::Code ProfilingHandler::handlerHeapProfiler(absl::string_view url, Http::Re
   return res;
 }
 
+Http::Code TcmallocProfilingHandler::handlerHeapDump(absl::string_view, Http::ResponseHeaderMap&,
+                                                     Buffer::Instance& response, AdminStream&) {
+  auto dump_result = Profiler::TcmallocProfiler::tcmallocHeapProfile();
+
+  if (dump_result.ok()) {
+    response.add(dump_result.value());
+    return Http::Code::OK;
```

Comment:
`admin.profile_path` is used as the output profile files' prefix for the gperftools. The gperftools will write profile results to the local file system automatically after the profiling was started.

But the new profiler, the heap result is only returned by every `/heap_dump` call. It can be write to any possition that callers want. So a pre-configured path is unnecessary here.